dolphinscheduler-task-executor/src/main/java/org/apache/dolphinscheduler/task/executor/container/AbstractTaskExecutorContainer.java

```diff
@@ -145,4 +150,26 @@ private void pushTaskExecutorLogToRemote(final ITaskExecutor taskExecutor) {
         }
     }
 
+    /**
+     * Delete task exec Directory if set development.state=false in common.properties explicit
+     * otherwise keep it for debug reason.
+     */
+    private void deleteExecPathIfProduction(ITaskExecutor taskExecutor) {
+        boolean isDev = PropertyUtils.getBoolean(Constants.DEVELOPMENT_STATE, true);
+        if (!isDev) {
+            try {
+                final TaskExecutionContext ctx = taskExecutor.getTaskExecutionContext();
+                final String execPath = ctx.getExecutePath();
+                if (StringUtils.isNotEmpty(execPath)) {
+                    FileUtils.deleteFile(execPath);
+                    log.info("Deleted task exec directory: {}", execPath);
+                }
+            } catch (Exception e) {
+                log.warn("Failed to delete task exec directory in prod mode.", e);
+            }
+        } else {
+            log.debug("Development mode is on, skip deleting executePath for debug reason");
+        }
+    }
```

Comment:
It's better to add a `finalize` method in `ITaskExecutor` to clear the `taskInstanceWorkingDirectory`.
We can add a method `clearTaskInstanceWorkingDirectory` in `TaskExecutionContextUtils` to do the real clear logic.

Reply:
Great, it's OK.